### DISCUSSION TITLE
基于Sui Groth16的ZK Demo

### DISCUSSION THREAD

**Discussion by MagicGordon** [NonceGeek/Move-Camp#287] [Amazing Things!]
以谜题板为思路，对sui的groth16模块进行实践，线下生成证明，链上进行验证。
主要功能：
1、创建谜题板：每个谜题板有自己固定的verifying_key
2、添加谜题：在指定谜题板上可以添加谜题以及答题奖励
3、答题：通过证明来获取对应谜题的答题奖励

**Comment by MagicGordon** [comment]
电路：https://github.com/MagicGordon/riddle_board/tree/main/circuit
合约：https://github.com/MagicGordon/riddle_board/tree/main/contract
前端：https://github.com/MagicGordon/riddle_board/tree/main/frontend

**Reply by baidang201** [reply]
大佬还缺人不，下个月学了zk后参与继续开发

**Reply by MagicGordon** [reply]
感谢关注，这个demo暂时不准备再深化了。
zk课程我也参与，学完可以一起构思更有意义的使用场景。

**Comment by leeduckgo** [comment]
👍
